sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1086,10 +1106,22 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   /**
    * Create a function database (optional) and name pair.
    */
-  protected def visitFunctionName(ctx: QualifiedNameContext): FunctionIdentifier = {
+  protected def visitFunctionName(
+      ctx: QualifiedNameContext,
+      trimFuncN: Option[String] = None): FunctionIdentifier = {
     ctx.identifier().asScala.map(_.getText) match {
-      case Seq(db, fn) => FunctionIdentifier(fn, Option(db))
-      case Seq(fn) => FunctionIdentifier(fn, None)
+      case Seq(db, fn) =>
+        if (fn.equalsIgnoreCase("trim") && trimFuncN.isDefined) {
```

Comment:
From `visitTrimFuncName`, `fn` can also be `ltrim` and `rtrim`, but seems you don't handle them here?